frontend/src/containers/Subscription/SubscriptionFeed.js

```diff
@@ -28,102 +28,112 @@ class SubscriptionFeed extends Component {
-                            } else {
-                                this.setState({
-                                    recoNum: 20,
-                                });
-                            }
-                        }
                     }).catch(() => {});
             }).catch(() => {});
     }
 
     clickMoreButton = () => {
+        // get new subscription
         if (!this.props.subscriptionFinished) {
             this.props.onGetSubscriptions({
                 page_number: this.props.subscriptionPageNum + 1,
             })
                 .then(() => {
-                    const { subscriptions } = this.state;
-                    this.setState({
-                        subscriptions: subscriptions.concat(this.props.subscriptionItems),
+                    this.setState((prevState) => ({
+                        subscriptions: prevState.subscriptions.concat(this.props.subscriptionItems),
+                        recoCount: 30 - this.props.subscriptionItems.length,
+                    }), () => {
+                        this.clickMoreButtonNext();
                     });
-                    this.clickMoreButtonNext();
                 }).catch(() => {});
         } else {
+            this.setState({
+                recoCount: 30,
+            });
             this.clickMoreButtonNext();
         }
     }
 
     clickMoreButtonNext = () => {
+        // get new recommendation
         if (!this.props.recommendationFinished) {
             this.props.onGetRecommendations({
                 page_number: this.props.recommendationPageNum + 1,
             })
                 .then(() => {
-                    const { recommendations } = this.state;
-                    this.setState({
-                        recommendations: recommendations.concat(this.props.recommendationItems),
+                    this.setState((prevState) => ({
+                        recommendations: prevState.recommendations.concat(
+                            this.props.recommendationItems,
+                        ),
+                    }), () => {
+                        this.addRecoToSub();
                     });
-                    this.addRecoToSub();
                 }).catch(() => {});
         } else {
             this.addRecoToSub();
         }
     }
 
     addRecoToSub = () => {
-        for (let i = this.state.recommendations.length - 1; i > this.state.start; i -= 1) {
+        // add new recommendation to subscription
+        const { recommendations } = this.state;
+        const { subscriptions } = this.state;
+        // sort recommendations
+        for (let i = recommendations.length - 1; i > this.state.start; i -= 1) {
             const j = Math.floor(Math.random() * (i + 1));
-            [this.state.recommendations[i], this.state.recommendations[j]] = [
-                this.state.recommendations[j],
-                this.state.recommendations[i],
+            [recommendations[i], this.state.recommendations[j]] = [
```

Comment:
Why is the only right `recommendations` from `this.state`?

Reply:
Wow Thank you for letting me knowㅠㅠ